sky/packages/sky/lib/src/rendering/editable_paragraph.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:sky/painting.dart';
+import 'package:sky/src/rendering/box.dart';
+import 'package:sky/src/rendering/object.dart';
+import 'package:sky/src/rendering/paragraph.dart';
+import 'package:sky/src/rendering/proxy_box.dart' show SizeChangedCallback;
+
+const _kCursorGap = 1.0;
+const _kCursorHeightOffset = 2.0;
+const _kCursorWidth = 1.0;
+
+/// A render object used by EditableText widgets.  This is similar to
+/// RenderParagraph but also renders a cursor and provides support for
+/// scrolling.
+class RenderEditableParagraph extends RenderParagraph {
+
+  RenderEditableParagraph({
+    TextSpan text,
+    Color cursorColor,
+    bool showCursor,
+    this.onContentSizeChanged,
+    Offset scrollOffset
+  }) : _cursorColor = cursorColor,
+       _showCursor = showCursor,
+       _scrollOffset = scrollOffset,
+       super(text);
+
+  Color _cursorColor;
+  bool _showCursor;
+  SizeChangedCallback onContentSizeChanged;
+  Offset _scrollOffset;
+
+  Size _contentSize;
+
+  Color get cursorColor => _cursorColor;
+  void set cursorColor(Color cursorColor) {
+    _cursorColor = cursorColor;
```

Comment:
In these setters, we usually check whether the old value is equal to the new value and skip the markNeedsPaint call in that case.  We do that because the widget layer generates unconditional writes into the render tree and we don't want to mark ourselves as needing paint when its not actually needed.